bundles/org.openhab.binding.openuv/src/main/java/org/openhab/binding/openuv/internal/handler/OpenUVReportHandler.java

```diff
@@ -158,28 +148,26 @@ private void updateChannels(ReportConfiguration config) {
     @Override
     public void dispose() {
         logger.debug("Disposing the OpenUV handler.");
-        ScheduledFuture<?> refresh = refreshJob;
-        if (refresh != null && !refresh.isCancelled()) {
-            refresh.cancel(true);
-        }
-        refreshJob = null;
 
-        ScheduledFuture<?> uxMax = uvMaxJob;
-        if (uxMax != null && !uxMax.isCancelled()) {
-            uxMax.cancel(true);
+        cancelFuture(refreshJob);
+        cancelFuture(uvMaxJob);
+    }
+
+    private void cancelFuture(@Nullable ScheduledFuture<?> job) {
+        if (job != null && !job.isCancelled()) {
+            job.cancel(true);
         }
-        uvMaxJob = null;
+        job = null;
```

Comment:
See my other comment: to be done rather by the caller.

Reply:
Done